lib/src/widget/arkit_scene_view.dart

```diff
@@ -740,4 +740,13 @@ class ARKitController {
     final result = await _channel.invokeMethod<Uint8List>('snapshot');
     return MemoryImage(result!);
   }
+
+  Future<Vector3?> cameraPosition() async {
+    final result = await _channel.invokeMethod('cameraPosition');
+    if(result != null) {
+      return _vector3Converter.fromJson(result!);
```

Comment:
`_vector3Converter.fromJson` expects a `List`, hence please use `invokeListMethod`. 

Dart is smart enough not to make you force unwrap the `result`. 

So that could be:
```dart
    final result = await _channel.invokeListMethod('cameraPosition');
    if(result != null) {
      return _vector3Converter.fromJson(result);
```

Would you please try it to check whether it still works?